pkg/apis/intevents/v1/topic_validation_test.go

```diff
@@ -105,6 +115,54 @@ func TestTopicCheckImmutableFields(t *testing.T) {
 			},
 			allowed: false,
 		},
+		"Project changed": {
+			orig: &topicSpec,
+			updated: TopicSpec{
+				Secret:  topicSpec.Secret,
+				Project: "new-project",
+				Topic:   topicSpec.Topic,
+			},
+			allowed: false,
+		},
+		"Secret.Key changed": {
+			orig: &topicSpec,
+			updated: TopicSpec{
+				Secret: &corev1.SecretKeySelector{
+					LocalObjectReference: corev1.LocalObjectReference{
+						Name: topicSpec.Secret.Name,
+					},
+					Key: "some-other-key",
+				},
+				Project: topicSpec.Project,
+				Topic:   topicSpec.Topic,
+			},
+			allowed: false,
+		},
+		"Secret.Name changed": {
+			orig: &topicSpec,
+			updated: TopicSpec{
+				Secret: &corev1.SecretKeySelector{
+					LocalObjectReference: corev1.LocalObjectReference{
+					Name: "some-other-name",
+					},
+					Key: topicSpec.Secret.Key,
+				},
+				Project: topicSpec.Project,
+				Topic:   topicSpec.Topic,
+			},
+			allowed: false,
+		},
+		"ServiceAccountName changed": {
```

Comment:
@nachocano Just want to confirm - in sources, we have ServiceAccountName should be immutable, so I think in topic and pullsubscription, it should be the same case. So I added some UT to verify that. Is that okay?

Reply:
yes, that's ok, thanks @danyinggu !